gamemodes/terrortown/gamemode/server/sv_roleselection.lua

```diff
@@ -556,7 +563,25 @@ local function SelectForcedRoles(plys, selectableRoles)
 		-- if it's not a selectable role, continue
 		if not roleCount then continue end
 
-		local curCount = 0
+		local isBaseRole = roles.GetByIndex(subrole):IsBaseRole()
+		local baserole = nil
+
+		--Consider maximum number of roles, that are available to the corresponding baserole
+		if not isBaseRole then
+			baserole = roles.GetByIndex(subrole).baserole
+
+			local baseroleCount = selectableRoles[baserole] - (selectedForcedRoles[baserole] or 0)
+
+			-- take the minimum of baseroles or subroles, if baseroles are available, else continue
+			if baseroleCount > 0 then
+				roleCount = math.min(baseroleCount, roleCount)
+			else
+				continue
+			end
+		end
+
+		-- Consider number of roles, that were forced by other subroles
+		local curCount = (selectedForcedRoles[subrole] or 0)
```

Comment:
Had to change this section here. Otherwise it was still possible to force more baseroles than available, depending on force order, when more subroles got chosen, than available baseroles.

This now always takes the minimum of baseroleCount and the subroleCount. As a maximum of available roles.
The currentCount just gets increased according to already forced subroles.

Seems to work, if not I probably have to come up with an automated testing, because the testing here is what takes most of the time. There are so many possibilities one doesn't know have an actually different behavior.